src/prof_data.c

```diff
@@ -374,6 +374,18 @@ prof_lookup(tsd_t *tsd, prof_bt_t *bt) {
 	return ret.p;
 }
 
+prof_tctx_t *
+prof_tctx_create(tsd_t *tsd) {
+	prof_tdata_t *tdata = prof_tdata_get(tsd, false);
+	if (tdata == NULL) {
```

Comment:
I don't think this can ever happen? We should have allocated tdata before reaching here.